broker/cloud_run/lsst/ps_to_storage/main.py

```diff
@@ -95,148 +84,62 @@ def run():
 def store_alert_data(envelope) -> None:
     """Uploads the msg data bytes to a GCP storage bucket."""
 
-    alert_bytes = base64.b64decode(envelope["message"]["data"])  # alert packet, bytes
-    attributes = envelope["message"].get("attributes", {})
-
-    # unpack the alert and read schema ID
-    header_bytes = alert_bytes[:5]
-    schema_id = deserialize_confluent_wire_header(header_bytes)
-
-    # get and load schema
-    schema = sr_client.get_schema(schema_id=schema_id)
-    parse_schema = json.loads(schema.schema_str)
-    schema_version = parse_schema["namespace"].split(".")[1]
-    content_bytes = io.BytesIO(alert_bytes[5:])
-
-    # deserialize the alert
-    alert_dict = fastavro.schemaless_reader(content_bytes, parse_schema)
-
-    # convert the MJD timestamp to "YYYY-MM-DD"
-    time_obj = Time(alert_dict["diaSource"]["midpointMjdTai"], format="mjd")
-    alert_date = time_obj.datetime.strftime("%Y-%m-%d")
-
-    filename = generate_alert_filename(
-        {
-            "schema_version": schema_version,
-            "alert_date": alert_date,
-            "objectId": alert_dict["diaObject"]["diaObjectId"],
-            "sourceId": alert_dict["diaSource"]["diaSourceId"],
-            "format": "avro",
-        }
-    )
+    # create an alert object from the envelope
+    alert = pittgoogle.Alert.from_cloud_run(envelope, "lsst")
 
-    blob = bucket.blob(filename)
-    blob.metadata = create_file_metadata(alert_dict, event_id=envelope["message"]["messageId"])
+    blob = bucket.blob(_generate_alert_filename(alert))
+    blob.metadata = _create_file_metadata(alert, event_id=envelope["message"]["messageId"])
 
     # raise a PreconditionFailed exception if filename already exists in the bucket using "if_generation_match=0"
     # let it raise. the main function will catch it and then drop the message.
-    blob.upload_from_string(alert_bytes, if_generation_match=0)
+    blob.upload_from_string(base64.b64decode(envelope["message"]["data"]), if_generation_match=0)
```

Comment:
I believe that the bytes needed here are in `alert.msg.data`. Also, let's move everything else out of this try/except.

```suggestion
    try:
        blob.upload_from_string(alert.msg.data, if_generation_match=0)
    except PreconditionFailed:
        # This alert is a duplicate. Drop it.
        return "", HTTP_204
```